scipy/ndimage/_morphology.py

```diff
@@ -220,7 +220,7 @@ def _binary_erosion(input, structure, iterations, mask, output,
     except TypeError as e:
         raise TypeError('iterations parameter should be an integer') from e
 
-    input = np.asarray(input)
+    input = np.asarray(input, order="C")
```

Comment:
```suggestion
    input = np.asarray(input)
```
Test is green after https://github.com/scipy/scipy/pull/22421/files goes in